### ISSUE TITLE
I find some error in googlenet.py ?

### ISSUE THREAD

**Issue by foreverfc1994** [pytorch/vision#906]
the kernel size should be 5 in  branch3,but there is 3
`self.branch3 = nn.Sequential(
            BasicConv2d(in_channels, ch5x5red, kernel_size=1),
            BasicConv2d(ch5x5red, ch5x5, kernel_size=3, padding=1)
        )`
Why?

**Comment by TheCodez** [comment]
The model was ported from tensorflow. For some reason they used 3x3 instead of the 5x5.

**Comment by oknuutti** [comment]
Maybe a bug in the tensorflow model? I think it should be 5x5. Maybe it's missing a second 3x3 convolutional layer? That would also result in an overall 5x5 convolution. I noticed that batch normalization is used, so based on e.g. [this post](https://towardsdatascience.com/a-simple-guide-to-the-versions-of-the-inception-network-7fc52b863202), the model here seems to be inception-v2 instead of the original GoogLeNet (inception-v1). In a way, a second 3x3 convolutional layer would then make more sense than just setting the kernel size to 5x5.

**Comment by TheCodez** [comment]
It’s most likely a bug, but it’s something that can’t easily be changed without retraining the model. If someone can change the 3x3 convolution to 5x5 and retrains the PyTorch model and gets the same or even a better accuracy that would be great, but that’s most likely not going to happen.

The Inception v2 model or also referred to as BN-Inception has more changes than just replacing one 5x5 with 2 consecutive 3x3 branches and adding batch normalization. So this is still the GoogLeNet model just with added batch normalization, which wasn’t available at the time of that paper so it was probably to speed up training.
Despite having to retrain the model when adding another 3x3 convolution, this also adds a computational cost of about 30% which is not desired for the GoogLeNet. I hope that clears some things up :)

**Comment by fmassa** [comment]
@TheCodez can you send a PR adding a comment for this discrepancy?

**Comment by yacineMahdid** [comment]
So if I understand correctly here, we are good with leaving that branch at as a 3x3 conv?
If it's a historical implementation seems to be more accurate to keep the 5x5 even though the performance is worse no?

Because if the use of that model is to be slightly more performant, we might as well drop all together one of the redundant 3x3 conv branch as it increases computational cost for not much.
